packages/loopring_v3/contracts/test/AmmPool.sol

```diff
@@ -204,17 +211,18 @@ contract AmmPool is IBlockReceiver, IAgent {
         }
     }
 
+    /// @param amounts The amounts to deposit + the amount of pool tokens to deposit
     function deposit(uint96[] calldata amounts)
         external
         payable
         online
     {
-        require(amounts.length == tokens.length, "INVALID_DATA");
+        require(amounts.length == tokens.length + 1, "INVALID_DATA");
 
         // Deposit the max amounts to this contract so we are sure
         // the amounts are available when the actual deposit to the exchange is done.
-        for (uint i = 0; i < tokens.length; i++) {
-            address token = tokens[i].addr;
+        for (uint i = 0; i < tokens.length + 1; i++) {
+            address token = (i < tokens.length) ? tokens[i].addr : address(this);
```

Comment:
if token == address(this), then the safeTransferFromAndVerify will fail, won't it?